src/lib/icon/icon.ts

```diff
@@ -198,11 +206,16 @@ export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, Can
   private _clearSvgElement() {
     const layoutElement: HTMLElement = this._elementRef.nativeElement;
     const childCount = layoutElement.childNodes.length;
+    const elementNodeType = this._document ? this._document.ELEMENT_NODE : 1;
```

Comment:
If this is the only thing document is needed for, I'm fine with just hard-coding `1` with a comment